### ISSUE TITLE
Plugin developers support for NS7

### ISSUE THREAD

**Issue by mrwrighty** [NativeScript/NativeScript#9180]
Nativescript relies heavily on 3rd party plugins to achieve certain functionality. The nativescript-google-maps-sdk for one and nativescript-videorecorder for another. I also used nativescript-background-geolocation-lt  extensively but the developer has dropped support for Nativescript because he doesn't see the platform going anywhere. The developer only supports Cordova/Ionic, React Native and Flutter. I moved from Ionic because Nativescript seemed to be going places.

I have used these in an app for a couple of years now, but these are broken in NS7 and there doesn't appear to be any plans from the developer to update it anytime soon.

I'm not sure what alternatives we have for the functions we require unless Nativescript themselves development the relevant plugins.

I currently have an app that I cannot upgrade with the functions I need because of this.

So So frustrating after investing a lot of time and effort in Nativescript.

**Comment by farfromrefug** [comment]
@mrwrighty under @nativescript-community we  have almost everything you need. Though it would require a bit of migration:
* @nativescript-community/gps
* @nativescript-community/ui-mapbox (mapbox sdk)
* @nativescript-community/ui-carto (carto mobile sdk)

And if i am correcty @triniwiz is still maintaining nativescript-videorecorder.

You can reach devs on slack, we are all there

**Comment by mrwrighty** [comment]
OK will take a look, @triniwiz add the enhancement label on Github back on 2 Oct 2020 but no activity since.

**Comment by farfromrefug** [comment]
@mrwrighty i will be glad to help in everything you need. Could also move the videorecorder plugin to the community so that it gets maintained again

**Comment by keerl** [comment]
I think @triniwiz is maintaining the videorecorder plugin in his monorepo. https://github.com/triniwiz/nativescript-plugins/tree/master/packages/nativescript-videorecorder

**Comment by triniwiz** [comment]
Update coming soon also I which part of the plugin do you mostly use the basic or advanced ?

**Comment by mrwrighty** [comment]
Very good news on the update. I use both basic and advanced depending on the app

**Comment by triniwiz** [comment]
Ah okay good to know 😄

**Comment by mrwrighty** [comment]
I haven't made the jump to TypeScript yet and find most of the demo's confusing when trying to translate the code to Javascript. Why is the assumption made that everyone is using TS and not JS when publishing new code, examples and demo content. I need the gps and ui-mapbox plugins to work in my JS app.

**Comment by triniwiz** [comment]
@mrwrighty u can grab it @ [@triniwiz/nativescript-videorecorder](https://triniwiz.github.io/nativescript-plugins/api-reference/video-recorder.html#installing)

**Comment by triniwiz** [comment]
One thing you should know is I moving the advanced module to another plugin

**Comment by mrwrighty** [comment]
> One thing you should know is I moving the advanced module to another plugin

Hi @triniwiz. I need the advanced functions. I have installed https://triniwiz.github.io/nativescript-plugins/api-reference/video-recorder.html#installing. Has the code changed significantly or will previous code work. 

Is this link for the advanced version.

**Comment by alexisconsuegra** [comment]
Hi @mrwrighty , I also find myself stuck with the dependencies. I am sorry I ask you but have you come across these warnings ?
Install packages.
npm WARN deprecated request@2.88.2: request has been deprecated, see request/request#3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

I can't find a way to correct them.

Thanks

**Comment by rigor789** [comment]
@alexisconsuegra My answer for a similar question on slack:
>  When you `npm i -g nativescript`, npm often prints out a bunch of deprecated dependencies, these are generally hard to avoid since they are often transitive dependencies deeply nested in the dependency tree.
>
> Some dependencies we are replacing with newer alternatives, but this is not a simple process, as parts of the CLI require rewrites to support alternative dependencies.
>
> In any case, these are not really a security concern, since they are not exposed to the public (ie. not running on a public facing server), and are only used on the local machine during development, where most of the security concerns don't apply.

**tl;dr** these are normal & nothing to worry about.

**Comment by alexisconsuegra** [comment]
I understand. Thanks.

**Comment by mrwrighty** [comment]
This has been closed but I do not see a fix for the issue.

ns plugin add @triniwiz/ nativescript-videorecorder results in 

npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /Users/nick/Documents/NativeScript/Projects/FRSWorkforce/@triniwiz/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/Users/nick/Documents/NativeScript/Projects/FRSWorkforce/@triniwiz/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/nick/.npm/_logs/2021-05-20T10_32_54_886Z-debug.log

I desperately need a fix for NS7/8 for the Videorecorder. I cannot find any other plugins to do the job.

**Comment by triniwiz** [comment]
`npm i @triniwiz/nativescript-videorecorder`

**Comment by mrwrighty** [comment]
My Code is 

`var appSettings = require("tns-core-modules/application-settings");
var dialogs = require("tns-core-modules/ui/dialogs");
var video_view_model_1 = require("./video-view-model");
//var nativescript_videorecorder_1 = require("nativescript-videorecorder");
var nativescript_videorecorder_1= require("@triniwiz/nativescript-videorecorder");
//var Advanced_1 = require("nativescript-fancy-camera");
var frame_1 = require("tns-core-modules/ui/frame");
var topmost = require("tns-core-modules/ui/frame").topmost;
var progress = require("tns-core-modules/ui/progress");
var page;
var recorder;
var interval;
var videoplayer;
var stoprecordingbutton;
var recordvideobutton;
var completesigninbutton;
var retakevideobutton;
var recorderstack;
var completestack;
var durationlabel;
var vm = new video_view_model_1.HelloWorldModel();
function navigatingTo(args) {

}
exports.navigatingTo = navigatingTo;

function loaded(args) {
    try{
    page = args.object;
    page.bindingContext = vm;
    vm.set('selectedVideo', '');
    appSettings.setString("vidfile","");
    appSettings.setNumber("duration", 0);
    Advanced_1.AdvancedVideoView.requestPermissions();
    recorder = args.object.page.getViewById("recorderView");
    vm.set('duration', recorder && recorder.duration ? recorder.duration : 0);
    // vm.set('duration', recorder && recorder.duration ? recorder.duration : appSettings.getNumber("duration"));
    recorder.on('started', args => {
        interval = setInterval(function() {
            vm.set('duration', recorder.duration);
//            if (recorder.duration > 9){ //Set to 44 for maximum 45 seconds of video
//                clearInterval(interval);
//                  // alert("You have reached the 45 second video limit");
//                stoprecordingbutton.notify({eventName: "tap", object: stoprecordingbutton});
//            }
        }, 1000);
    });

    recorder.on('finished', function(args){
        clearInterval(interval);
        var vidfile = args.object.get('file');
        //alert(vidfile);
        page.bindingContext.set('selectedVideo', args.object.get('file'));
        appSettings.setString("vidfile", vidfile);
        durationlabel.text = "Video Length "+recorder.duration+" Secs";
    });
    videoplayer = args.object.page.getViewById("videoplayer");
    videoplayer.className="hideObject";
    recorder.className="showObject";
    stoprecordingbutton = args.object.page.getViewById("stoprecordingbutton");
    recordvideobutton = args.object.page.getViewById("recordvideobutton");
    retakevideobutton = args.object.page.getViewById("retakevideobutton");
    durationlabel = args.object.page.getViewById("durationlabel");
    durationlabel.class = "showObject";
    completestack = args.object.page.getViewById("completestack");
    completestack.className="hideObject";
    recorderstack = args.object.page.getViewById("recorderstack");
    recorderstack.className="showObject";
    }
    catch(Err){
        alert("Loaded Error: "+Err);
    }
}
exports.loaded = loaded;

function recordVideo(){
    var recorder = page.getViewById("recorderView");
    videoplayer.className="hideObject";
    recordvideobutton.isEnabled=false;
    recordvideobutton.className="fadeObject";
    recorder.startRecording();
}
exports.recordVideo = recordVideo;

function retakeVideo(args){
    var recorder = page.getViewById("recorderView");
    vm.set('duration', 0);
    appSettings.setNumber("duration",0);
    recorder.className="showObject";
    videoplayer.className="hideObject";
    recorderstack = args.object.page.getViewById("recorderstack");
    completestack = args.object.page.getViewById("completestack");
    durationlabel = args.object.page.getViewById("durationlabel");
    //durationlabel.class = "showObject";
    //durationlabel.class="videotime";
    durationlabel.text = "Video Length 0 Secs";
    recorderstack.className="showObject";
    completestack.className="hideObject";
    page.bindingContext.set('selectedVideo', '');
    videoplayer.className="hideObject";
    recordvideobutton.isEnabled=true;
}
exports.retakeVideo = retakeVideo;

function stopRecording(){
    try{
    var recorder = page.getViewById("recorderView");
    recorderstack = page.getViewById("recorderstack");
    completestack = page.getViewById("completestack");
    //durationlabel = args.object.page.getViewById("durationlabel");
    //durationlabel.text = "Video Length "+recorder.duration+" Secs";
    //durationlabel.class = "hideObject";
    recordvideobutton.isEnabled = true;
    recordvideobutton.className="unfadeObject";
    retakevideobutton = page.getViewById("retakevideobutton");
    retakevideobutton.isEnabled = true;
    recorder.className="hideObject";
    videoplayer.className="showObject";
    recorderstack.className="hideObject";
    completestack.className="showObject";
    appSettings.setNumber("duration", recorder.duration);
    recorder.stopRecording();
}
    catch(err){
        alert("An error occured recording the video. Please minimise the app (DO NOT CLOSE COMPLETLEY) by pressing the home button and re-open the app and try again.");
    }
}
exports.stopRecording = stopRecording;

function goToVideoRecorder(event){
    frame_1.goBack();
}
exports.goToVideoRecorder = goToVideoRecorder;

function saveVideo(){
    var vidfile = appSettings.getString("vidfile");
    if (vidfile == ""){
       dialogs.confirm({
    title: "Warning",
    message: "You must take a video of the vehicle to complete the sign in process!",
    okButtonText: "OK",
  });
        return;
    }
    const movetosigninloc = {
        moduleName: "signinloc/signinloc-page",
        animated: true,
        clearHistory: false,
        transition: {
          name: "flip",
          duration: 380,
          curve: "easeIn"
        }
      };
      frame_1.topmost().navigate(movetosigninloc);
}
exports.saveVideo = saveVideo;
`

My Page definition is:-

`<Page navigatingTo="navigatingTo" class="page" xmlns="http://schemas.nativescript.org/tns.xsd" xmlns:VideoPlayer="nativescript-videoplayer" xmlns:recorder="@triniwiz/nativescript-videorecorder" loaded="loaded">
	<ActionBar title="Vehicle Video" class="action-bar">
	</ActionBar>
    <GridLayout rows="*,100,50,60">
        <recorder:AdvancedVideoView row="0" quality="qvga" width="90%" cameraPosition="back" id="recorderView" fill="true" thumbnailCount="1" duration="20" marginTop="10"/>

        <VideoPlayer:Video row="0" width="90%" src="{{selectedVideo}}" loaded="videoplayerLoaded" finished="videoFinished" autoplay="false" id="videoplayer" marginTop="20"/>
        <Label row="1" text="Video must show any defects reported in the checks." textAlignment="center" width="90%" textWrap="true" class="redLabel" />
	<StackLayout row="2">
            <Label id="durationlabel" textAlignment="center" text="{{'Video Length '+duration+' Secs'}}" class="videotime" row="3"/>

        </StackLayout>
        <StackLayout row="3" orientation="horizontal" backgroundColor="#333333" horizontalAlignment="center" width="100%" id="recorderstack">

            <Button textWrap="true" width="50%" textAlignment="center" backgroundColor="#333333" color="#ffffff" tap="stopRecording" id="stoprecordingbutton">
            <FormattedString>
                <Span text="Stop Video"></Span>
            </FormattedString>
        </Button>
            <Button textWrap="true" width="50%" textAlignment="center" backgroundColor="#333333" color="#ffffff" tap="recordVideo" id="recordvideobutton">
            <FormattedString>
                <Span text="Record Video"></Span>
            </FormattedString>
        </Button>
        </StackLayout>
        <StackLayout row="3" orientation="horizontal" backgroundColor="#333333" horizontalAlignment="center" width="100%" id="completestack">
            <Button textWrap="true" width="50%" textAlignment="center" backgroundColor="#333333" color="#ffffff" tap="saveVideo" id="completesigninbutton">
            <FormattedString>
                <Span text="Complete Sign In"></Span>
            </FormattedString>
        </Button>
            <Button textWrap="true" width="50%" textAlignment="center" backgroundColor="#333333" color="#ffffff" tap="retakeVideo" id="retakevideobutton">
            <FormattedString>
                <Span text="Retake Video"></Span>
            </FormattedString>
        </Button>
		</StackLayout>



	</GridLayout>
</Page>
`
Since moving to @triniwiz/nativescript-videorecorder, the page will not load.

**Comment by triniwiz** [comment]
1. Why do u have `tns-core-modules` imports if this is v7
2. I removed the `AdvancedVideoView` to move it into another plugin > https://github.com/NativeScript/NativeScript/issues/9180#issuecomment-771670567

**Comment by mrwrighty** [comment]
I have tried @triniwiz/nativescript-fancy-camera but still have the same issue. Page will not load.

**Comment by triniwiz** [comment]
I never said it was done if you want a custom camera atm feel free to use camplus

**Comment by mrwrighty** [comment]
My apologies I thought the project was complete. If I remove the AdvanceVideoView should this fix the issue.
